.github/scripts/deploy_dbt_model_dependencies.sh

```diff
@@ -1,184 +1,155 @@
 #!/usr/bin/env bash
-# Parse, bundle, and upload dbt Python model requirements to S3.
+# Parse, install, zip, and upload dbt Python model requirements to S3.
 #
-# Takes one or more optional positional arguments representing the models for
-# which you would like to upload dependencies. For example, the following
-# command will only upload dependencies for the `reporting.ratio_stats` model:
+# This script checks for Python model dependencies by searching every
+# model node for a `config.packages` attribute and uploading a zip archive
+# for every version of every package that is specified in one or more such
+# attributes. It expects attribute values to be well-formed Python requirement
+# specifiers that are pinned to a specific version of the package, e.g.
+# `assesspy==1.0.1` is allowed but not `assesspy==1.0.*` or `assesspy~=1.0.1`.
 #
-# ../.github/scripts/deploy_dbt_model_dependencies.sh reporting.ratio_stats
+# If a zip archive for a given version of a dependency is already present in
+# the remote dependency dir on S3, the script will skip installing, zipping,
+# and uploading that dependency version.
 #
-# When no arguments are provided, the script will upload dependencies for all
-# models:
+# When uploading dependencies, package names get cleaned in order to make them
+# fit the required format for importable Python identifiers. This process
+# involves the following transformations:
 #
-# ../.github/scripts/deploy_dbt_model_dependencies.sh
+# * Hyphens ("-") and periods (".") get replaced with underscores ("_")
+# * Double equals signs ("==") get replaced with the version prefix "_v"
+#
+# For example, the string "assesspy==1.0.1" will get saved to a zip archive
+# called "assesspy_v1_0_1". This string can then be used to import the package
+# into a Python model script with a set of import calls like so:
+#
+#   sc.addPyFile(f"{s3_dependency_dir}/assesspy_v1_1_0.zip")
+#   import assesspy_v1_1_0 as assesspy
 #
 # Assumes that dbt, python, and jq are installed and available on the caller's
 # path. Also assumes that it is being run in a directory containing a
 # dbt project.
+#
+# Example usage:
+#
+#   ../.github/scripts/deploy_dbt_model_dependencies.sh
 set -euo pipefail
 
 # Fall back to local development if a dbt $TARGET is not specified
 target=${TARGET:-"dev"}
 echo "Parsing dbt Python model dependencies for target '$target'"
 
 # Determine the remote location where bundled dependencies will be deployed
-s3_dependency_dir=$(dbt run-operation print_s3_dependency_dir --quiet \
-    --target "$target")
-
-# Parse optional positional arguments representing a restricted list of
-# models for which to upload dependencies
-declare -a specified_models
-if [ $# -gt 0 ]; then
-    specified_models=("$@")
-fi
```

Comment:
Since packages are now configured in a global repository rather than in environments isolated by model and dbt target, it felt like `specified_models` and its positional args were adding unnecessary complexity, so I went ahead and removed it. We may want to reintroduce this logic to speed up build times if the universe of packages we use for our scripts ends up growing by a lot, but in the meantime I think reduced maintenance complexity is more important than a slight reduction in build times. Let me know if you disagree!